### ISSUE TITLE
Complex objects - how to use `localUri` to upload a file?

### ISSUE THREAD

**Issue by ffxsam** [awslabs/aws-mobile-appsync-sdk-js#63]
In the complex object documentation, I see that we're supposed to create an object including a string property `localUri` that points to the file, and this is how AppSync will upload to S3.

The problem is, I don't understand how this could possibly work. First, when using HTML `<input type="file">`, the local file path is not provided when the user chooses a file (for security reasons). Secondly, the data used to upload a file is a `File` blob, which one can use `FileReader` to read the binary data and send to S3 in the `Body` param. But `localUri` doesn't accept a `File` blob.

So I'm confused as to how this works, exactly.

**Comment by manueliglesias** [comment]
Hi @ffxsam 

thanks for trying this feature!

### TL;DR:
Pass the value of your `<input type="file">` as the `localUri` property.

### Long version
Whatever you have in `localUri` will be passed as the `Body` to the [upload method in the S3 client](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property) from the aws-sdk. This can be of types:
>`Body — (Buffer, Typed Array, Blob, String, ReadableStream) Object data.`

When working with files in the browser (`<input type="file">`), you work with a [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) object which conveniently is:
> A `File` object is a specific kind of a `Blob`, and can be used in any context that a `Blob` can. 

The aws-appsync sdk does this in the [src/link/complex-object-link-uploader.js](https://github.com/awslabs/aws-mobile-appsync-sdk-js/blob/5fa3dadab2678539340cb2f8e18fc8fbf61bc3e8/packages/aws-appsync/src/link/complex-object-link-uploader.js#L17)

**Comment by ffxsam** [comment]
Ahh I see. So then I'd need to use `FileReader` to read the `File` blob into an array buffer, and pass that array buffer as the `localUri` property. Got it.

The only thing missing here is the ability to hook into the `s3.upload` `httpUploadProgress` event.

**Comment by M1chaelTran** [comment]
Has anyone else able to upload multiple files?
Single file upload seems to work... but not for multiple files.

my mutation and input types below
```
mutation($input: [UploadFileInput!]!){
  uploadFiles(input: $input){
    items{
      ...FileInfo
    }
  }
}

input UploadFileInput {
  file: S3ObjectInput
  name: String
}

input S3ObjectInput {
  bucket: String
  key: String
  region: String
  mimeType: String
  localUri: String
}
```
